app/perfaddresscache.h

```diff
@@ -42,12 +42,13 @@ class PerfAddressCache
 
     struct SymbolCacheEntry
     {
-        SymbolCacheEntry(quint64 offset = 0, quint64 value = 0, quint64 size = 0, const QByteArray &symname = {}, const QByteArray &mangledSymName = {})
+        SymbolCacheEntry(quint64 offset = 0, quint64 value = 0, quint64 size = 0, const QByteArray &symname = {}, const QByteArray &mangledSymName = {}, quint64 adjust = 0)
```

Comment:
I wonder, this patch message sounds pretty generic and not directly related to the disassembler - I think this should be split off into a separate review request and then merged. ideally, can you write a test case for this too?

please also describe the issue that this is solving in the commit message. I believe it means we are computing wrong function location addresses for functions that are missing debug symbols? i.e. those where we don't find a cu die or sub program die?

Reply:
The problem concerns kernel symbols from *.ko modules. The modules are relocated. 
My test for this is perf.data recorded with sudo rights on 'coremark' test.

Reply:
OK, so is it possible to create a test for this? Afaik a normal library is also getting relocated when compiled with PIE, no? So ideally we could even test this without kernel modules? I guess we simply never noticed this, as we don't use the addresses yet in the GUI consumers of the perfparser output. So if you'd output them in `PerfParserTestClient::convertToText` and then in a follow up patch apply this change, then we should hopefully see a difference?

Reply:
Ok, I will try